src/Components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+type Props = {
+  todos: Todo[];
+  onDelete: (todoId: number) => void;
+  isLoading: number[];
+  tempTodo: Todo | null;
+  toggleTodo: (todoToUpdate: Todo) => void;
+  updateTodo: (updatedTodo: Todo) => Promise<void>;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  onDelete,
+  isLoading,
+  tempTodo,
+  toggleTodo,
+  updateTodo,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {/* This is a completed todo */}
+      {todos &&
+        todos.map(todo => (
+          <TodoItem
+            key={todo.id}
+            todo={todo}
+            onDelete={onDelete}
+            isLoading={isLoading.includes(todo.id)}
+            toggleTodo={toggleTodo}
+            updateTodo={updateTodo}
+          />
+        ))}
+
+      {tempTodo && (
+        <TodoItem
+          todo={tempTodo}
+          onDelete={() => {}}
+          isLoading={isLoading.includes(0)}
```

Comment:
```suggestion
          isLoading
```
temp todo always should be displayed with loader
So we also can skip adding 0 to `isLoading` array in the App.tsx